contracts/0.4.24/StETHPermit.sol

```diff
@@ -0,0 +1,175 @@
+// SPDX-FileCopyrightText: 2023 OpenZeppelin, Lido <info@lido.fi>
+
+// SPDX-License-Identifier: GPL-3.0
+
+/* See contracts/COMPILERS.md */
+pragma solidity 0.4.24;
+
+import {ECDSA} from "../common/lib/ECDSA.sol";
+
+import {StETH} from "./StETH.sol";
+
+/**
+ * @dev Interface of the ERC20 Permit extension allowing approvals to be made via signatures, as defined in
+ * https://eips.ethereum.org/EIPS/eip-2612[EIP-2612].
+ *
+ * Adds the {permit} method, which can be used to change an account's ERC20 allowance (see {IERC20-allowance}) by
+ * presenting a message signed by the account. By not relying on {IERC20-approve}, the token holder account doesn't
+ * need to send a transaction, and thus is not required to hold Ether at all.
+ */
+interface IERC2612 {
```

Comment:
>Can value some explainer why don't we use the standard OpenZeppelin interface here, but the custom one.

There is no OZ version existing with {I}ERC2612 for Solidity 0.4.24 :eyes:

Reply:
>And we can probably move it to the separate file, since it is a full-fledged independent interface, not an excerpt, but it's discussable.

I guess it's not worth doing since the interface isn't Lido-specific enough and can be imported from OZ or composed easily by hand in place.